### DISCUSSION TITLE
Where to put Interfaces

### DISCUSSION THREAD

**Discussion by nadjitan** [nuxt/nuxt#15986] [Questions] [answered]
I wanted to know where to properly put interfaces because I have IUser in files `components/User.vue` and `server/api/users.ts`.  
```ts
interface IUser {
  id: number
  name: string
  ...
}
```

Is there a good way to avoid duplicates of interface?

**Comment by Diizzayy** [comment, marked as answer]
@Kapatid You can create a directory called `types` in the root of your nuxt3 project and add declaration files as needed.

`types/user.d.ts`
```ts
declare interface IUser {
  id?: number;
  name?: string;
}
```

You should now be able to access the interface in your components or else where in your app
```ts
<script lang="ts" setup>
const user = ref<IUser>(null)
</script>
```

`user` should now be fully typed as seen below.
![Example In IDE](https://user-images.githubusercontent.com/19627670/154400420-1c5f3ce6-dd84-461d-a43c-ad337930562b.png)

**Reply by nadjitan** [reply]
Thanks!

**Reply by happycoder80** [reply]
I tried this using the latest Nuxt3 but VSCode still does not find my types. Am I missing something obvious?

**Reply by nadjitan** [reply]
@happycoder80  I am using `"nuxt3": "latest"` in my devDependencies but It still works for me so I am not sure why it does not work for you maybe vscode just needs a restart.

**Comment by maxdzin** [comment]
Hi folks!

I've tried to put declarations inside `types` or `@types` directories. Neither works for Nuxt v3.0.0-rc2 in VS Code.
Has anyone found a nice workaround for this?

**Comment by yashsamit** [comment]
It is still an open problem for me. I will appreciate any solution!

**Reply by mihail727** [reply]
Did the above method not work for you?
Just to clarify we're only talking about `interfaces` and `types`?
Just if we are also talking about `classes` and `enums` then such a thing will not work with them (as far as I know)

**Reply by mihail727** [reply]
Also, do not forget that you cannot use ordinary imports inside the declared file if you suddenly need them.

**Reply by yashsamit** [reply]
Thanks, it helped! I solved problem using your trick i.e. I removed all import statements from d.ts files inside types folder.
